apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/rest/service/TransportInteractionServiceImpl.java

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright © 2018-2019 Apollo Foundation
+ */
+
+package com.apollocurrency.aplwallet.apl.core.rest.service;
+
+import com.apollocurrency.aplwallet.api.response.TransportStatusResponse;
+import com.apollocurrency.aplwallet.apl.util.injectable.PropertiesHolder;
+import java.net.URI;
+import java.net.URISyntaxException;
+import javax.inject.Inject;
+import javax.inject.Singleton;
+import lombok.Setter;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+@Singleton
+public class TransportInteractionServiceImpl implements TransportInteractionService {
+    
+    
+    private static final Logger log = LoggerFactory.getLogger(TransportInteractionServiceImpl.class);
+    private String wsUrl; 
+
+    private TransportInteractionWebSocket transportInteractionWebSocket;
+    
+    @Setter
+    private boolean done;
```

Comment:
done is not thread safe. Use volatile keyword to guarantee synchronized read/write for this value.